sklearn/mixture/gmm.py

```diff
@@ -501,10 +523,33 @@ def fit(self, X, y=None):
             self.covars_ = best_params['covars']
             self.means_ = best_params['means']
             self.weights_ = best_params['weights']
+        else:
```

Comment:
Indeed we need to cover that case with an explicit test if it's not already covered.